File: app/src/main/java/sabalan/paydar/mohtasham/ezibazi/controller/system/hardware/ConnectivityListener.java
package sabalan.paydar.mohtasham.ezibazi.controller.system.hardware;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.widget.Toast;

import sabalan.paydar.mohtasham.ezibazi.controller.system.application.G;
import sabalan.paydar.mohtasham.ezibazi.view.activity.ActivityNetworkError;

public class ConnectivityListener extends BroadcastReceiver {




  private Snackbar snackbar;
  private View view;
  private boolean isShowNetworkError = false;

//  public ConnectivityListener(View view, Snackbar snackbar) {
//    this.view = view;
//    this.snackbar = snackbar;
//  }
//
//  public ConnectivityListener(View view) {
//    this.view = view;
//  }

  public ConnectivityListener(){

  }


  @Override
  public void onReceive(Context context, Intent intent) {
    ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
    assert connectivityManager != null;
    NetworkInfo networkInfo=connectivityManager.getActiveNetworkInfo();
    boolean isConnected = networkInfo!=null && networkInfo.isConnected();


    if(isConnected){
//      if(snackbar != null) {
//        snackbar.dismiss();
//      }

      if (G.mustReconnect){
        G.mustReconnect = false;
        Intent i = G.context.getPackageManager().getLaunchIntentForPackage( G.context.getPackageName() );
        assert i != null;
        i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        context.startActivity(i);
        isShowNetworkError = false;

      }







    }else{

        G.mustReconnect = true;
        if(!isShowNetworkError) {
          Intent intent1 = new Intent(context, ActivityNetworkError.class);
          context.startActivity(intent1);
          isShowNetworkError = true;
        }



//        snackbar = Snackbar.make(view, "خطا در اتصال به سرور", Snackbar.LENGTH_INDEFINITE);
//        snackbar.show();
    }
  }
  
}

File: app/src/main/java/ir/easybazi/system/hardware/ConnectivityListener.kt
package ir.easybazi.system.hardware

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.view.View
import com.google.android.material.snackbar.Snackbar

import ir.easybazi.system.application.G
import ir.easybazi.view.activity.ActivityNetworkError

//  public ConnectivityListener(View view, Snackbar snackbar) {
//    this.view = view;
//    this.snackbar = snackbar;
//  }
//
//  public ConnectivityListener(View view) {
//    this.view = view;
//  }

class ConnectivityListener : BroadcastReceiver() {


    private val snackbar: Snackbar? = null
    private val view: View? = null
    private var isShowNetworkError = false
    private lateinit var context: Context


    override fun onReceive(context: Context, intent: Intent) {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val networkInfo = connectivityManager.activeNetworkInfo
        val isConnected = networkInfo != null && networkInfo.isAvailable && networkInfo.isConnected && networkInfo.isConnectedOrConnecting


        if (isConnected) {
            //      if(snackbar != null) {
            //        snackbar.dismiss();
            //      }

            if (G.mustReconnect) {
                G.mustReconnect = false
                val i = G.context.packageManager.getLaunchIntentForPackage(G.context.packageName)!!
                i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                context.startActivity(i)
                isShowNetworkError = false

            }


        } else {

            G.mustReconnect = true
            if (!isShowNetworkError) {
                val intent1 = Intent(context, ActivityNetworkError::class.java)
                context.startActivity(intent1)
                isShowNetworkError = true
            }


            //        snackbar = Snackbar.make(view, "خطا در اتصال به سرور", Snackbar.LENGTH_INDEFINITE);
            //        snackbar.show();
        }
    }

}
